src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -498,6 +498,36 @@ public boolean hasSchemaEvolutionPermission(String tableName, String role) {
     return hasPermission;
   }
 
+  /**
+   * Alter iceberg table to modify columns datatype
+   *
+   * @param tableName the name of the table
+   * @param columnInfosMap the mapping from the columnNames to their infos
+   */
+  @Override
+  public void alterColumnsDataTypeIcebergTable(
+      String tableName, Map<String, ColumnInfos> columnInfosMap) {
+    LOGGER.debug("Modifying data types of iceberg table columns");
+    String alterSetDatatypeQuery = generateAlterSetDataTypeQuery(columnInfosMap);
+    executeStatement(tableName, alterSetDatatypeQuery);
+  }
+
+  private String generateAlterSetDataTypeQuery(Map<String, ColumnInfos> columnsToModify) {
+    StringBuilder setDataTypeQuery = new StringBuilder("alter iceberg ");
+    setDataTypeQuery.append("table identifier(?) alter column ");
+    for (Map.Entry<String, ColumnInfos> column : columnsToModify.entrySet()) {
+      String columnName = column.getKey();
+      String dataType = column.getValue().getColumnType();
+
+      setDataTypeQuery.append(columnName).append(" set data type ").append(dataType).append(", ");
+    }
+    // remove last comma and whitespace
+    setDataTypeQuery.deleteCharAt(setDataTypeQuery.length() - 1);
```

Comment:
I think you can even try to avoid deleting chars by utilizing Collectors.joining:
```
    String columnsPart = columnsToModify.entrySet().stream().map(column -> {
      String columnName = column.getKey();
      String dataType = column.getValue().getColumnType();
      return columnName + " set data type " + dataType;
    }).collect(Collectors.joining(", "));
    
    setDataTypeQuery.append(columnsPart);
```

Reply:
Yea it's more elegant. I apply it.